### PR TITLE
Update to DEVELOPMENT-SNAPSHOT-08-18-a

### PR DESCRIPTION
**Pull request by rymcol**
This commit updates core syntax to the 08-18-a development snapshot of swift.

### CODE REVIEW

File: Sources/Core/Portal.swift [hunk 1]
```diff
@@ -17,11 +17,11 @@ public enum PortalError: Error {
      This class is designed to make it possible to use asynchronous contexts in a synchronous environment.
 */
 public final class Portal<T> {
-    private var result: Result<T>? = .none
+    fileprivate var result: Result<T>? = .none
```
Comment (tanner0101) [inline Sources/Core/Portal.swift]: I think we can just move this to internal (aka remove access keyword)

**Comment by loganwright** [comment]
@tannernelson why the move from fileprivate on the portal's result?

I think it's important to protect this type as user should never be setting it directly. Doing so w/o adequately closing the portal would at best be unexpected behavior.

I'd like to see the fileprivate reintroduced to the type.

**Comment by loganwright** [comment]
Looks great, let's get tests passing and integrate this. Some of the pthread cancel stuff seems to be breaking down on Linux for some reason.

File: Sources/Core/Portal.swift [hunk 2]
```diff
@@ -17,11 +17,11 @@ public enum PortalError: Error {
      This class is designed to make it possible to use asynchronous contexts in a synchronous environment.
 */
 public final class Portal<T> {
-    private var result: Result<T>? = .none
+    var result: Result<T>? = .none
```
Comment (loganwright) [inline Sources/Core/Portal.swift:20]: This should move to fileprivate, also a comment on main convo.

**Comment by tanner0101** [comment]
Going to merge this and update on a branch we own.

**Pull request merged**